pkg/services/ngalert/notifier/receiver_svc.go

```diff
@@ -271,26 +279,162 @@ func (rs *ReceiverService) DeleteReceiver(ctx context.Context, uid string, orgID
 		if err != nil {
 			return err
 		}
-		return rs.deleteProvenances(ctx, orgID, postable.GrafanaManagedReceivers)
+		return rs.deleteProvenances(ctx, orgID, existing.Integrations)
+	})
+}
+
+func (rs *ReceiverService) CreateReceiver(ctx context.Context, r *models.Receiver, orgID int64, user identity.Requester) (*models.Receiver, error) {
+	if err := rs.authz.AuthorizeCreate(ctx, user, r); err != nil {
+		return nil, err
+	}
+
+	revision, err := rs.cfgStore.Get(ctx, orgID)
+	if err != nil {
+		return nil, err
+	}
+
+	createdReceiver := r.Clone()
+	err = createdReceiver.Encrypt(rs.encryptor(ctx))
+	if err != nil {
+		return nil, err
+	}
+
+	err = revision.CreateReceiver(&createdReceiver)
+	if err != nil {
+		return nil, err
+	}
+	createdReceiver.Version = createdReceiver.Fingerprint()
+
+	err = rs.xact.InTransaction(ctx, func(ctx context.Context) error {
+		err = rs.cfgStore.Save(ctx, revision, orgID)
+		if err != nil {
+			return err
+		}
+		return rs.setReceiverProvenance(ctx, orgID, &createdReceiver)
+	})
+	if err != nil {
+		return nil, err
+	}
+	return &createdReceiver, nil
+}
+
+func (rs *ReceiverService) UpdateReceiver(ctx context.Context, r *models.Receiver, storedSecureFields map[string][]string, orgID int64, user identity.Requester) (*models.Receiver, error) {
+	if err := rs.authz.AuthorizeUpdate(ctx, user, r); err != nil {
```

Comment:
I think here is the problem: we authorize access to receiver with UID(name1) and rename name1 -> name2, which effectively changes the UID of the resulting receiver. 

authorization should detect whether user has access to old and new UID.